tests/integration/targets/dns_config/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+
+  block:
+    - name: Add/modify configs (search domains, dns servers)
+      scale_computing.hypercore.dns_config:
+        search_domains:
+          - "test"
+          - "test1"
+          - "test2"
+          - "test3"
+          - "test4"
+        dns_servers:
+          - "1.2.3.4"
+        state: present
+      register: result
+    - ansible.builtin.assert:
+        that:
+          - result.diff.before != result.diff.after
```

Comment:
Check also result.records is correct, and same for result.changed. Those two are usually used later in playbook.

The result.diff is usually used only for debugging - when you look into verbose output on your screen, so most other integration tests do not check in depth what is inside (the != check is a bonus, so please keep it).